app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -511,6 +511,36 @@ public function getCartURL()
         return $this->boltHelper()->getMagentoUrl('checkout/cart');
     }
 
+    /**
+     * Returns the One Page or Multi-Page checkout Publishable key depending on the requested
+     * page and which key has been set in the admin.  Cart and Product pages prioritize the multi-step key
+     * and falls back to the payment only key.  All other pages prioritize the payment only key and falls back
```

Comment:
What is the reason for all other pages prioritizing the payment only key

Reply:
Wouldn't you want all pages on a storefront to prioritize multi-step

Reply:
I agree with @ryan-timothy-albert here the intuition is to prioritize multi-step as that's our core focus.

Reply:
@ryan-timothy-albert and @matt-thomason 
While at first pass, it seems that multi-step should be our default, and of course I was thinking that while writing this code, it is very important to ask oneself questions to like, 

_**"hmm... the code originally is defaulting to one-page/payment only.  I wonder why this was writtenin  that way and what would be the consequences of me changing this default and deploying it to existing merchants?"**_

https://github.com/BoltApp/bolt-magento1/blob/4eecb3edf0fd0c1571bf66f360e10681e763c556/app/design/frontend/base/default/template/boltpay/track.phtml#L31-L33

**So, I am going to weigh heavy on the fact that the logic is longstanding in production without issue before trying to change that logic and deploy it to our current merchants.**

If we wish to explore changing this logic, I move that we pursue it in a different pull request where the risk and impact are properly assessed as changing this logic is not the focus of this PR.  **For now, I believe that it should suffice that if no payment-only publishable key is issued to the merchant, then the multi-step key will always be used which is in alignment with our core focus.**